doc/md/writing-motoko/candid.md

```diff
@@ -0,0 +1,135 @@
+
+At its core, Candid provides a language-agnostic way to describe and transmit data.
+It ensures strong typing, which means data is correctly interpreted across different services and languages.
+This type safety is complemented by an efficient binary format for encoding data, making it ideal for network transmission.
+In the context of Motoko, Candid is deeply integrated into the language.
+Motoko automatically generates Candid interfaces for canister smart contracts, and provides built-in functions like `to_candid` and `from_candid` for easy serialization and
+deserialization of data to and from Candid format.
+
+On the broader Internet Computer platform, Candid serves as the standard communication protocol between canisters. When one canister calls another, the arguments are serialized to Candid, transmitted, and then deserialized by the receiving canister. This standardization enables developers to create frontends in languages like JavaScript that can easily interact with backend canisters written in Motoko or Rust.
+
+The IC SDK includes various tools for working with Candid, such as utilities for generating interface files and creating TypeScript bindings from Candid definitions.
+These tools further enhance the developer experience and promote interoperability.
+
+Importantly, Candid's design allows for backwards-compatible upgrades of canister interfaces.
+This feature facilitates the evolution of services over time, a critical aspect for long-lived applications
+on the Internet Computer.
+
+In essence, Candid forms a fundamental part of the IC ecosystem. It provides a standardized way to describe, serialize, and communicate data between different parts of the platform. Its seamless integration with Motoko and other languages makes it easier for developers to create interoperable and type-safe services on the Internet Computer.
+
+## Explicit Candid serialization
+
+Motoko's `to_candid` and `from_candid` operators allow you to work with Candid-encoded data.
+
+`to_candid (<exp1>, ..., <expn>)` serializes a sequence of Motoko values into a `Blob` containing a Candid binary encoding of the data.
+
+For example,
+
+``` motoko no-repl
+let encoding : Blob = to_candid ("dogs", #are, ['g', 'r', 'e', 'a', 't']);
+```
+
+`from_candid <exp>` deserializes a blob that contains Candid data back into a Motoko value.
+
+``` motoko no-repl
+ let ?(t, v, cs) = from_candid encoding : ?(Text, {#are; #are_not}, [Char]);
+```
+
+`from_candid` will trap if its arguments is a blob that does not contain valid Candid data.
+Because deserialization can fail if the encoded value does not have the expected Candid type,
+`from_candid` returns a value of option type, with `null` indicated the encoding is well-formed but of the wrong Candid type or some value `?v`
+where `v` is the decoded value.
+`from_candid` can only be used in the context of other code that determines its optinal result type, for which a type annotation may be required.
```

Comment:
```suggestion
Because deserialization can fail if the encoded value does not have the expected Candid type, `from_candid` returns a value of option type, with `null` indicating the encoding is well-formed but of the wrong Candid type or some value `?v`, where `v` is the decoded value. `from_candid` can only be used in the context of other code that determines its optional result type, for which a type annotation may be required.
```